src/libraries/Common/src/System/IO/Compression/ZLibNative.cs

```diff
@@ -262,81 +262,101 @@ private void EnsureState(State requiredState)
             }
 
 
-            public ErrorCode DeflateInit2_(CompressionLevel level, int windowBits, int memLevel, CompressionStrategy strategy)
+            public unsafe ErrorCode DeflateInit2_(CompressionLevel level, int windowBits, int memLevel, CompressionStrategy strategy)
             {
                 EnsureNotDisposed();
                 EnsureState(State.NotInitialized);
 
-                ErrorCode errC = Interop.zlib.DeflateInit2_(ref _zStream, level, CompressionMethod.Deflated, windowBits, memLevel, strategy);
-                _initializationState = State.InitializedForDeflate;
+                fixed (ZStream* stream = &_zStream)
+                {
+                    ErrorCode errC = Interop.zlib.DeflateInit2_(stream, level, CompressionMethod.Deflated, windowBits, memLevel, strategy);
+                    _initializationState = State.InitializedForDeflate;
 
-                return errC;
+                    return errC;
+                }
             }
 
 
-            public ErrorCode Deflate(FlushCode flush)
+            public unsafe ErrorCode Deflate(FlushCode flush)
             {
                 EnsureNotDisposed();
                 EnsureState(State.InitializedForDeflate);
-                return Interop.zlib.Deflate(ref _zStream, flush);
+
+                fixed (ZStream* stream = &_zStream)
+                    return Interop.zlib.Deflate(stream, flush);
```

Comment:
Nit: can you use braces for these one liners, too?

Reply:
Done.